File: 02.producto_escalar/python/producto_escalar.py
# -*- coding: utf-8 -*-
"""
Desarrolle una función que calcule el producto escalar de dos vectores.
"""


def main():
    u = (3, 0, 0)  # X, Y, Z
    v = (5, 5, 0)  # X, Y, Z
    r = producto_escalar(u, v)
    print(r)


def producto_escalar(u, v):
    ''' El producto escalar de dos vectores se calcula con la siguiente ecuación:
            u x v = u1 * v1 + u2 * v2 + u3 * v3
        :param
            u y v - Tuplas de tres elementos que representa (x, y, z) de un vector
        :return
            Número real
    '''
    assert isinstance(u[0], int), "El vector de entrada no cumple el formato correcto"
    assert isinstance(u[1], int), "El vector de entrada no cumple el formato correcto"
    assert isinstance(u[2], int), "El vector de entrada no cumple el formato correcto"
    assert isinstance(v[0], int), "El vector de entrada no cumple el formato correcto"
    assert isinstance(v[1], int), "El vector de entrada no cumple el formato correcto"
    assert isinstance(v[2], int), "El vector de entrada no cumple el formato correcto"

    return u[0] * v[0] + u[1] * v[1] * u[2] * v[2]


if __name__ == "__main__":
    main()

File: 02.producto_escalar/java/productoEscalar.java
/*
Desarrolle una función que calcule el producto escalar de dos vectores.
*/
public class productoEscalar {
    public static void main(String[] args) {
        Vector u = new Vector(3, 0, 0);
        Vector v = new Vector(5, 5, 0);
        int resultado = u.productoEscalar(v);
        System.out.println("Resultado: " + resultado);
                
    }
        
}

/*
Clase vector
*/
class Vector {
    int x = 0;
    int y = 0;
    int z = 0;
    
    Vector(int x, int y, int z) {
        this.x = x;
        this.y = y;
        this.z = z;
    }
    
    /* Cálculo del producto escalar */
    public int productoEscalar(Vector u) {
        int r = this.x * u.x + this.y * u.y + this.z * u.z;
        return r;
    }
    
}
